src/charm.py

```diff
@@ -56,6 +59,11 @@ def __init__(self, *args):
         # Gathers scrape job information from metrics endpoints
         self.metrics_consumer = MetricsEndpointConsumer(self, "metrics-endpoint")
 
+        # Exposes remote write endpoints
+        self.remote_write = PrometheusRemoteWriteProvider(
+            self, relation_name="remote-write-server"
+        )
```

Comment:
`self.remote_write_provider.set_endpoint` (with the `_provider` suffix) seems less confusing to me.

Reply:
I agree